src/tests/scroll_factor.rs

```diff
@@ -0,0 +1,315 @@
+use niri_config::{FloatOrInt, Mouse, Touchpad};
+
+// Reproduce the macro from input/mod.rs for testing
+macro_rules! extract_scroll_factors {
```

Comment:
After you put the scroll-factor config into a `struct ScrollFactor` and add a method on that struct, these tests can and should move to niri-config.